### PR TITLE
allow multiple delete network modifications

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java [hunk 1]
```diff
@@ -77,11 +78,11 @@ public void handleExcludeModification(AbstractNode node, UUID modificationUuid,
     }
 
     @Override
-    public void removeModificationToExclude(AbstractNode node, UUID modificationUuid) {
+    public void removeModificationToExclude(AbstractNode node, List<UUID> modificationUuid) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java:81]: Typo ? : removeModificationsToExclude(AbstractNode node, List<UUID> modificationsUuids)

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java [hunk 2]
```diff
@@ -563,11 +563,11 @@ public void doHandleExcludeModification(UUID nodeUuid, UUID modificationUUid, bo
     }
 
     @Transactional
-    public void doRemoveModificationToExclude(UUID nodeUuid, UUID modificationUUid) {
+    public void doRemoveModificationToExclude(UUID nodeUuid, List<UUID> modificationUUid) {
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java]: fixed

File: src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java [hunk 3]
```diff
@@ -563,11 +563,11 @@ public void doHandleExcludeModification(UUID nodeUuid, UUID modificationUUid, bo
...
         nodesRepository.findById(nodeUuid).ifPresent(n -> repositories.get(n.getType()).removeModificationToExclude(nodeUuid, modificationUUid));
     }
 
-    public Mono<Void> removeModificationToExclude(UUID nodeUuid, UUID modificationUuid) {
+    public Mono<Void> removeModificationToExclude(UUID nodeUuid, List<UUID> modificationUuid) {
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java]: fixed

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java [hunk 4]
```diff
@@ -76,7 +76,7 @@ public UUID getSecurityAnalysisResultUuid(AbstractNode node) {
     public void handleExcludeModification(AbstractNode node, UUID modificationUuid, boolean active) {
     }
 
-    public void removeModificationToExclude(AbstractNode node, UUID modificationUuid) {
+    public void removeModificationToExclude(AbstractNode node, List<UUID> modificationUuid) {
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java]: fixed

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java [hunk 5]
```diff
@@ -178,7 +178,7 @@ public void handleExcludeModification(UUID nodeUuid, UUID modificationUuid, bool
         handleExcludeModification(getNode(nodeUuid), modificationUuid, active);
     }
 
-    public void removeModificationToExclude(UUID nodeUuid, UUID modificationUuid) {
+    public void removeModificationToExclude(UUID nodeUuid, List<UUID> modificationUuid) {
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java]: fixed

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java (hunk 1 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java]: fixed

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java [hunk 6]
```diff
@@ -77,11 +78,11 @@ public void handleExcludeModification(AbstractNode node, UUID modificationUuid,
     }
 
     @Override
-    public void removeModificationToExclude(AbstractNode node, UUID modificationUuid) {
+    public void removeModificationsToExclude(AbstractNode node, List<UUID> modificationUuid) {
```
Comment (niconoir) [inline src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java:81]: Typo: modification**s**Uuid**s** ?

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java [hunk 7]
```diff
@@ -77,11 +78,11 @@ public void handleExcludeModification(AbstractNode node, UUID modificationUuid,
...
         NetworkModificationNode networkModificationNode = (NetworkModificationNode) node;
         if (networkModificationNode.getModificationsToExclude() != null &&
-            networkModificationNode.getModificationsToExclude().contains(modificationUuid)) {
-            networkModificationNode.getModificationsToExclude().remove(modificationUuid);
+            networkModificationNode.getModificationsToExclude().containsAll(modificationUuid)) {
```
Comment (niconoir) [inline src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java:84]: modificationsToExclude list will not necessarily contain all modificationUuid of the `modificationUuid` list. This test is not correct. Maybe check **for each modification to remove** if it is present in the modificationsToExclude list. Or better, determine the intersection of the modificationsToExclude and modificationUuid lists and remove elements of this list.

**Review by niconoir — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/NetworkModificationService.java [hunk 8]
```diff
@@ -293,15 +294,13 @@ private String buildPathFrom(UUID networkUuid) {
...
+    public Mono<Void> deleteModifications(UUID groupUuid, List<UUID> modificationsUuids) {
         Objects.requireNonNull(groupUuid);
-        Objects.requireNonNull(modificationUuid);
-        var path = UriComponentsBuilder.fromPath(GROUP_PATH
-                + DELIMITER + "modifications" + DELIMITER + "{modificationUuid}")
-            .buildAndExpand(groupUuid, modificationUuid)
-            .toUriString();
+        Objects.requireNonNull(modificationsUuids);
+        var path = UriComponentsBuilder.fromPath(GROUP_PATH + DELIMITER + "modifications");
+        modificationsUuids.forEach(m -> path.queryParam("modificationsUuids", modificationsUuids));
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/NetworkModificationService.java:301]: doesn't work unfortunately :(

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java (hunk 6 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java]: fixed

File: src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java (hunk 7 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/networkmodificationtree/NetworkModificationNodeInfoRepositoryProxy.java]: if the test was only to check presence before removing it, it is useless, we search 2 times removed test

File: src/main/java/org/gridsuite/study/server/NetworkModificationService.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/NetworkModificationService.java:318]: If it really does not work, at least change `m -> path.queryParam("modificationsUuids", modificationsUuids)` to `m -> path.queryParam("modificationsUuids", m)`.

**Review by FranckLecuyer — APPROVED** [review]

**Review by niconoir — APPROVED** [review]

**Pull request merged**